### PR TITLE
docs: Use data folder for website carousel

### LINKED ISSUE #2055: Use /data for the Hugo website carousel
See: https://github.com/spf13/hugo/blob/master/docs/layouts/index.html#L142

This should be a list of "quote objects" in /data.

Two reasons for this:

1. We should eat our own dog food
2. Building the docs is an important CI test, so we should strive to use the core Hugo features.

### PR DESCRIPTION
**Pull request by robertbasic**
Fixes #2055

### CODE REVIEW

**Comment by bep** [comment]
That was quick! This works, and I should have been more specific, but I think this would be better as a single file, lets call it  `/data/references.toml` (we can add other stuff to it later):

```toml
 +[[quotes]]
+name = "Baron Schwartz"
+twitter_handle = "@xaprb"
+quote = "Hugo is impressively capable. It's a static site generator by <a href='https://twitter.com/spf13'> @spf13 </a> written in <a href='https://twitter.com/hashtag/golang?src=hash'> #golang </a> . Just upgraded to latest release; very powerful.  "
+link = "https://twitter.com/xaprb/status/556894866488455169"
+date = "2015-01-18"
+[[quotes]]
+name = "Dave Cottlehuber"
+twitter_handle = "@dch__"
+quote = "I just fell in love with #hugo, a static site/blog engine written by @spf13 in #golang  + stellar docs"
+link = "https://twitter.com/dch__/status/460158115498176512"
+date = "2014-04-26"
```
And so on ... This way we get an array that we can do sorting with etc. And it will remove one range from the template.

**Comment by robertbasic** [comment]
@bep done. Not sure how you meant to remove one range from the template. By not having a separate partial for the quotes?

File: docs/layouts/index.html [hunk 1]
```diff
@@ -140,38 +140,13 @@
...
-                      <div class="item"> <blockquote><p>Hugo is impressively capable. It's a static site generator by <a href="https://twitter.com/spf13">@spf13</a> written in <a href="https://twitter.com/hashtag/golang?src=hash">#golang</a>. Just upgraded to latest release; very powerful.</p>&mdash; Baron Schwartz (@xaprb) <a href="https://twitter.com/xaprb/status/556894866488455169">January 18, 2015</a></blockquote> </div>
-                      <div class="item"> <blockquote><p><a href="https://twitter.com/hashtag/Hugo?src=hash">#Hugo</a> A very good alternative for <a href="https://twitter.com/hashtag/wordpress?src=hash">#wordpress</a>!!! A fast and modern static website engine <a href="http://gohugo.io/">gohugo.io</a></p>&mdash; WorkHTML (@workhtml) <a href="https://twitter.com/workhtml/status/563064361301053440">February 4, 2015</a></blockquote> </div>
+                      {{ range $.Site.Data.references }}
```
Comment (bep) [inline docs/layouts/index.html]: I'll merge this as this is perfectly fine for now. What I meant by removing one range what the above. Or replace it with 

```
 {{ partial "quote.html" ($.Site.Data.references.quotes}}
```

But that isn't needed until we get more reference types.

Comment (bep) [inline docs/layouts/index.html]: Oh, and the partial should be named quotes.html I guess, to be pedantic. As it expects a list of quotes.

Comment (robertbasic) [inline docs/layouts/index.html]: Ah yes, I could pass the data in like that as well. I'm here to learn :)

**Comment by robertbasic** [comment]
@bep don't merge then yet, I'll fix it up real quick!

**Comment by bep** [comment]
Good!

**Comment by bep** [comment]
That looks good. Merged in 40f62a8.

**Pull request closed without merging**